File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans(nums.size(), 1);
        int val = 1;
        for(int i = 0; i < nums.size(); i++){
            ans[i] = val;
            val*=nums[i];
        }
        int val2 = 1;
        for(int i = nums.size()-1; i >= 0; i--){
            ans[i]*=val2;
            val2*=nums[i];
        }
        return ans;
    }
};

File: Data Structures & Algorithms/squares-of-a-sorted-array/submission-0.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left = 0
        right = len(nums)-1
        ans = []
        while left <= right:
            r = nums[right]
            l = nums[left]
            if r*r > l*l:
                ans.append(r*r)
                right-=1
            else:
                ans.append(l*l)
                left+=1
        return ans[::-1]


        